dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultJobSchedulerService.java

```diff
@@ -99,6 +101,22 @@ public void executeNow(@Nonnull String jobId) throws NotFoundException, Conflict
     }
   }
 
+  @Override
+  @Transactional
+  public void revertNow(@Nonnull UID jobId)
+      throws ConflictException, NotFoundException, ForbiddenException {
+    UserDetails currentUser = getCurrentUserDetails();
+    if (currentUser == null || !currentUser.isAuthorized("F_PERFORM_MAINTENANCE"))
+      throw new ForbiddenException(JobConfiguration.class, jobId.getValue());
```

Comment:
FYI: This is what I mentioned in the weekly meeting, if I move the authority check into this service I can get away with not having to load the configuration. Or if I would have needed to I could have evicted it as the scope here is just this operation. But I decided that it is good enough that `F_PERFORM_MAINTENANCE` and super can perform the operation.